### ISSUE TITLE
ERROR: Package 'python-vehichle-simulator' requires a different Python: 3.8.8 not in '>=3.10'

### ISSUE THREAD

**Issue by huisky2000** [cybergalactic/PythonVehicleSimulator#2]
Hi,
I tried to install the package on virtual environment. However, it keeps showing the same error even I tried to installed it under virtual environment with specific python version=3.8.8.
Anyone know how to solve this?
Regards
Huisky

**Comment by cybergalactic** [comment]
The "match" statement in main.py requires  Python >= 3.10. You can rewrite this using if statements. Remember to change to 3.8.8 in setup.cfg as well. Alternatively, you must install 3.10.

**Comment by huisky2000** [comment]
Hi again,
Thanks a lot for the prompt reply. I probably misunderstood the error message in the beginning. Anyway, I tried to build it in the virtual environment with python 3.10.4. It was successfully installed after I run 'pip install .\', however, when I run the 'main.py' file, it says 'ModuleNotFoundError: No module named 'python_vehicle_simulator'.
Tried a little bit working around. it seems following packages are installed:

pip list
Package                   Version
------------------------- -------
atomicwrites              1.4.0
attrs                     21.4.0
colorama                  0.4.4
cycler                    0.11.0
fonttools                 4.33.3
iniconfig                 1.1.1
kiwisolver                1.4.2
matplotlib                3.5.2
numpy                     1.22.4
packaging                 21.3
Pillow                    9.1.1
pip                       22.1.2
pluggy                    1.0.0
py                        1.11.0
pyparsing                 3.0.9
pytest                    7.1.2
python-dateutil           2.8.2
python-vehichle-simulator 1.0.1
setuptools                62.3.2
six                       1.16.0
tomli                     2.0.1
wheel                     0.37.1

note that the name is 'python-vehichle-simulator', not 'python_vehichle_simulator'.
if I try to import 'python-vehichle-simulator', it is not allowed by python :(

>>> import python-vehicle-simulator
  File "<stdin>", line 1
    import python-vehicle-simulator
                 ^
SyntaxError: invalid syntax

so, what's wrong here?

Regards
Huisky

**Comment by cybergalactic** [comment]
This works on my computer:
     Click on the green code tab on Github and choose download
     This gives you the catalog PythonVehicleSimulator-master 

python3 -m pip install -e /MY_PATH/PythonVehicleSimulator-master

Then pip list should give you: python-vehichle-simulator 1.0.1

**Comment by huisky2000** [comment]
it is strange. Are you able to import  it with 'hyphens' in the name?
yes, it is listed as you said (see above), but still the name are different from the 'main.py'.

According to here:
"Identifiers cannot contain hyphens. This is not a question of style, but a part of the language syntax, see http://docs.python.org/reference/lexical_analysis.html#identifiers"

**Comment by cybergalactic** [comment]
The package is imported using the path + downloaded file catalog name, that is   /MY_PATH/PythonVehicleSimulator-master.  You can also rename the file catalog. 

Do not use, pip install python_vehichle_simulator or pip install python-vehicle-simulator. 
This should resolve the problem.

python3 -m pip install -e /MY_PATH/PythonVehicleSimulator-master

**Comment by huisky2000** [comment]
ok, some progress at least. the importing is now working.
i run 'pytest -k simulate -v'
got following:
tests\test_simulate.py:83: NameError
=============================================== short test summary info ===============================================
ERROR tests/test_simulate.py::test_simulate - NameError: name 'simulate' is not defined
ERROR tests/test_simulate.py::test_plot_vehicle_states - NameError: name 'simulate' is not defined
ERROR tests/test_simulate.py::test_plot_controls - NameError: name 'simulate' is not defined
ERROR tests/test_simulate.py::test_3D_plot - NameError: name 'simulate' is not defined
FAILED tests/test_simulate.py::test_DSRV - NameError: name 'simulate' is not defined
FAILED tests/test_simulate.py::test_otter - NameError: name 'simulate' is not defined
FAILED tests/test_simulate.py::test_ROVzefakkel - NameError: name 'simulate' is not defined
FAILED tests/test_simulate.py::test_semisub - NameError: name 'simulate' is not defined
FAILED tests/test_simulate.py::test_shipClarke83 - NameError: name 'simulate' is not defined
FAILED tests/test_simulate.py::test_supply - NameError: name 'simulate' is not defined
FAILED tests/test_simulate.py::test_frigate - NameError: name 'simulate' is not defined
FAILED tests/test_simulate.py::test_tanker - NameError: name 'simulate' is not defined
FAILED tests/test_simulate.py::test_remus100 - NameError: name 'simulate' is not defined

but it seems working if i run 'main.py' file directly under 'src' folder :(

**Comment by cybergalactic** [comment]
My mistake. I have changed the structure of the file catalogs. Then I forgot to update the imports in test_simulate.py. They should be:

import matplotlib.pyplot as plt
from python_vehicle_simulator.lib import *
import python_vehicle_simulator.vehicles as vehicles
import pytest

I have updated the file on GitHub. No the tests work again. 
Remember to run pytest -k simulate -v from the catalog  /MY_PATH/PythonVehicleSimulator
